src/main/java/com/solopov/urlshorter/dao/entities/generators/id/ShortUrlGenerator.java

```diff
@@ -0,0 +1,20 @@
+package com.solopov.urlshorter.dao.entities.generators.id;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.hibernate.HibernateException;
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+import org.hibernate.id.IdentifierGenerator;
+
+import java.io.Serializable;
+import java.util.Locale;
+
+public class ShortUrlGenerator implements IdentifierGenerator {
+
+    private final static Integer SHORT_LINK_SIZE = 4;
```

Comment:
nit:
it's not a link size per se but rather URL_KEY_LENTH

Reply:
upd commit: changed